tests/relay/test_relay_dispatcher.py

```diff
@@ -33,6 +36,18 @@ def get_dispatcher():
     return dispatcher
 
 
+async def _attempt_timestamp(event_builder, timestamps: list[int], should_save: bool):
```

Comment:
I can't really understand what the function is doing with the `attempt_timestamp`. can we rename it?